MarkBind/markbind
# Add progress indicator feature

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] New feature




**What is the rationale for this request?**
This PR adds the progress indicator as outlined in #983. This gives authors the choice to add a reading progress indicator to pages.

**What changes did you make? (Give an overview)**
Added `.progress` CSS class in `markbind.css` that styles the progress bar. 
Added logic in `setup.js` to get the scroll percentage in a webpage.
In order to enable the progress bar in the pages, the following markup can be added to the header file:
```html
<div class="progress"></div>
```

**Provide some example code that this change will affect:**


```js
function setupReadingProgress() {
  const h = document.documentElement;
  const b = document.body;
  const st = 'scrollTop';
  const sh = 'scrollHeight';
  const progress = document.querySelector('.progress');
  let scroll;

  document.addEventListener('scroll', () => {
    scroll = Number(h[st] || b[st]) / Number((h[sh] || b[sh]) - h.clientHeight);
    scrollPercentage = scroll * 100;
    progress.style.setProperty('--scroll', `${scrollPercentage}%`);
  });
}
```


**Testing instructions:**
* Run `markbind init` 
* Add the markup to the header file (`_markbind/headers/header.md`)
* Run `markbind serve` to generate the site
* Scroll to see the progress indicator

**Proposed commit message: (wrap lines at 72 characters)**


Add reading progress indicator
